backend/src/shortener/shortener.controller.spec.ts

```diff
@@ -27,64 +40,82 @@ describe('ShortenerController', () => {
 
   describe('shortener', () => {
     it('should return a shortern url', async () => {
-      jest.spyOn(shortenerService, 'generateShortUrl').mockReturnValue('best');
-      jest.spyOn(shortenerService, 'isShortUrlAvailable').mockResolvedValue(true);
+      jest.spyOn(shortenerService, 'generateShortenedUrl').mockReturnValue('best');
+      jest.spyOn(shortenerService, 'isShortenedUrlAvailable').mockResolvedValue(true);
       jest.spyOn(shortenerService, 'addUrl').mockResolvedValue(undefined);
-      jest.spyOn(shortenerService, 'isUrlAlreadyShortend').mockReturnValue(false);
+      jest.spyOn(shortenerService, 'isUrlAlreadyShortened').mockReturnValue(false);
 
-      const body: ShortenerDTO = { originalUrl: 'https://github.com/origranot/reduced.to' };
-      const short = await shortenerController.shortener(body);
+      const body: ShortenerDto = { originalUrl: 'https://github.com/origranot/reduced.to' };
+      const req = {} as any as Request;
+      const short = await shortenerController.shortener(body, req);
       expect(short).toStrictEqual({ newUrl: 'best' });
     });
 
     it('should throw an error of invalid url', () => {
-      const body: ShortenerDTO = { originalUrl: 'invalid-url' };
+      const body: ShortenerDto = { originalUrl: 'invalid-url' };
+      const req = { user: { id: 'user_id' } } as any as Request;
       expect(async () => {
-        await shortenerController.shortener(body);
+        await shortenerController.shortener(body, req);
       }).rejects.toThrow(BadRequestException);
     });
 
+    it('should throw an error if the original URL is already shortened', async () => {
+      jest.spyOn(shortenerService, 'isUrlAlreadyShortened').mockReturnValue(true);
+      const body: ShortenerDto = { originalUrl: 'https://github.com/origranot/reduced.to' };
+      const req = { user: { id: 'user_id' } } as any as Request;
+
+      try {
+        await shortenerController.shortener(body, req);
+        throw new Error('Expected an error to be thrown!');
+      } catch (err) {
+        expect(err.message).toBe('The URL is already shortened...');
+      }
+    });
```

Comment:
I guess I did it and you just added the req object. this is a weird test tbh.
Just to keep in my mind to change this.